### PR TITLE
Fix 'https' scheme in ES hosts setting when set from Cloud ID.

### PR DESCRIPTION
**Pull request by guyboertje**
This is a blocker for the combination of x-pack feature config management and or monitoring when using cloud id or specifying host(s) urls in logstash.yml

I did get 2 wait timeout converge_spec.rb failures on my laptop - only when running rake test:core though, the individual spec file passed.

Jenkins tests pass.

### CODE REVIEW

**Comment by tbragin** [comment]
@guyboertje @acchen97 @andrewvc @jordansissel Just checking to see if this is still on track to get resolved in RC2?

File: logstash-core/spec/logstash/util/cloud_setting_id_spec.rb [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    it '#kibana_host has a value' do
+      expect(subject.kibana_host).to eq("identifier.us-east-1.aws.found.io:443")
+    end
+    it '#kibana_scheme has a value' do
+      expect(subject.kibana_scheme).to eq("https")
+    end
+    it '#to_s has a value of #decoded' do
+      expect(subject.to_s).to eq(subject.decoded)
+    end
+  end
```
Comment (andrewvc) [inline logstash-core/spec/logstash/util/cloud_setting_id_spec.rb]: These specs all seem to cover unacceptable input. What about acceptable input?

Comment (guyboertje) [reply logstash-core/spec/logstash/util/cloud_setting_id_spec.rb]: line 67 should read `describe "when given acceptable input, the accessors:" do`, sorry - will fix.

**Comment by guyboertje** [comment]
@andrewvc 
How about now?
I added a kibana_client test too.

**Comment by guyboertje** [comment]
yay

**Pull request merged**